services/QuillLMS/app/controllers/snapshots_controller.rb

```diff
@@ -77,6 +60,32 @@ def options
     return render json: { error: 'user is not authorized for all specified schools' }, status: 403
   end
 
+  private def retrieve_cache_or_enqueue_worker(worker)
+    return { error: 'unrecognized query type for this endpoint' } unless worker::QUERIES.keys.include?(snapshot_params[:query])
```

Comment:
This is an HTTP 200, by default. I think we want this error to be less silent - perhaps use `ErrorNotifier`?

Alternatively you could fire off an error from the client, but pushing logic from server to client seems unnecessary.

Reply:
I agree with the point about using ErrorNotifier

Reply:
I like the idea of returning some non-200 status code here.  It'll make it easier for the front-end to know something went wrong.  Let me see if I can figure out how to make that work.